### PR TITLE
Leddie24/component bug fixes

### PR DESCRIPTION
**Pull request by leddie24**
This PR contains bug fixes for some components as well as documentation updates.  Below are a list of changes, as well as screenshots with before and after, if applicable. 

### List of changes

- Changed forEach to for loop, in localhost documentation.  Fixes bug in Firefox.
- Fixed PersonaCard missing the script file, so the example component wasn't functioning correctly
- Fixed TextField component not removing the label/placeholder when a user tabs to it, rather than clicks.  I changed the eventlistener from "click" to "focus"
- Fixed SearchBox-icon incorrect height in input, as well as duplicate rules.   Removing the height: 16px seems to fix this issue.

**BEFORE**
icon-before

**AFTER**
icon-after

- Fixed the textfield component placeholder being cut off in the PeoplePicker documentation.  I changed the parent container to have a width of 100% instead of auto, as well as adding a text-overflow: ellipsis as a fallback.

**BEFORE**
textfield-before

**AFTER**
textfield-after

- I added in JS in the component documentation for Panel
- I changed the padding for documentation from the content container, to the componentDemo class instead, so the Overlay component will display correctly

**BEFORE**
padding-before

**AFTER**
padding-after

-  I fixed incorrect line-height on the Dropdown component.  Selected and disabled options had a line height of 40px.  This addresses issue #78 

**BEFORE**
dropdown-bug

**AFTER**
dropdown-after

- Fixed ContextualMenu's text overflow bug ([Issue #65 ](https://github.com/OfficeDev/office-ui-fabric-js/issues/65)) I added a max-width of 220px (matching the class --hasIcons's max width), and added text-overflow ellipsis to the menu items. 

**BEFORE**
contextualmenu-before

**AFTER**
contextualmenu-after

- Fixes issues in #27.  Panel and overlay originally had position absolute on them, which resulted in incorrect position when the user isn't in the very top of the page.  Position Fixed solves this issue.

- Updated several components missing MDL2 icons  (DatePicker, FacePile, ListItem, MessageBanner, PeoplePicker, Table, Callout, Pivot)

- Added example JS to button component documentation
- Fix pivot menuitem being misaligned.

**BEFORE**
pivot-before

**AFTER**
pivot-after

### CODE REVIEW

**Comment by battletoilet** [comment]
Approved

File: src/components/Overlay/Overlay.scss [hunk 1]
```diff
@@ -27,3 +27,7 @@
 .ms-Overlay--dark {
   background-color: $ms-color-blackTranslucent40;
 }
+
+.ms-noScroll {
```
Comment (mikewheaton) [inline src/components/Overlay/Overlay.scss]: This should probably be a utility class (`.ms-u-noScroll` or `.ms-u-overflowHidden`), although that would mean moving it over to Fabric Core. Or should we have component-specific utilities? Honestly it's probably not worth the trouble right now, maybe just rename the class and leave it in this file.

File: src/components/Table/Table.scss [hunk 2]
```diff
@@ -40,13 +40,8 @@
 
       // But show the mark.
       &::after {
-        @include ms-Icon;
-        content: '\e041';
+        content: '\e73A';
```
Comment (mikewheaton) [inline src/components/Table/Table.scss]: The pattern we're following now is to use the Sass mixin to fill in this content. So you'd want to do `@include ms-Icon--CheckboxComposite` and that will add `content: '\E73A'.

File: src/components/Overlay/Overlay.scss (hunk 1 above)
Comment (leddie24) [inline src/components/Overlay/Overlay.scss]: Thanks @mikewheaton.  I wasn't aware of the -u- prefix as a utility class.  I'll rename and recommit now.

File: src/components/Table/Table.scss [hunk 3]
```diff
@@ -99,15 +82,14 @@
...
   // Empty checkbox.
-  &::before {
-    border: 1px solid $ms-color-neutralTertiary;
-    content: '';
-    display: block;
-    height: 14px;
-    left: 2px;
+  &::after {
+    @include ms-Icon;
+    content: '\e739';
```
Comment (mikewheaton) [inline src/components/Table/Table.scss]: Same here, try using `@include ms-Icon--Checkbox` to bring in the content value.

**Comment by mikewheaton** [comment]
Approved. I added a few small suggestions but overall this looks really good.

**Pull request merged**